server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -352,6 +364,61 @@ private CompletionStage<RestResponse> report(RestRequest restRequest) {
       );
    }
 
+   private CompletionStage<RestResponse> nodeReport(RestRequest restRequest) {
+      EmbeddedCacheManager cacheManager = invocationHelper.getProtocolServer().getCacheManager();
+      List<Address> members = cacheManager.getMembers();
+      if (members == null) {
+         return report(restRequest);
+      }
+
+      String targetName = restRequest.variables().get("nodeName");
+      ByRef<ByteBuffer> response = new ByRef<>(null);
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      return SecurityActions.getClusterExecutor(cacheManager)
+            .submitConsumer(ecm -> {
+               CacheManagerInfo cmi = ecm.getCacheManagerInfo();
+               if (!cmi.getNodeName().equals(targetName)) return null;
+
+               GlobalComponentRegistry gcr = SecurityActions.getGlobalComponentRegistry(ecm);
+               BasicComponentRegistry bcr = gcr.getComponent(BasicComponentRegistry.class);
+               ServerStateManager ssm = bcr.getComponent(ServerStateManager.class).wired();
+
+               CompletionStage<byte[]> report = ssm.managedServer().getServerReport()
+                     .thenApply(path -> {
+                        try {
+                           return Files.readAllBytes(path);
+                        } catch (IOException e) {
+                           throw new CacheException(e);
+                        }
+                     });
+               return CompletableFutures.uncheckedAwait(report.toCompletableFuture());
+            }, (ignore, info, t) -> {
+               if (t != null) {
+                  throw CompletableFutures.asCompletionException(t);
+               }
+
+               if (info != null) {
+                  response.set(ByteBuffer.wrap(info));
+               }
+            })
+            .thenApply(ignore -> {
+               ByteBuffer report = response.get();
+               if (report == null) {
+                  return responseBuilder.status(NOT_FOUND).build();
```

Comment:
I don't think this is the correct response. Shouldn't this be a `INTERNAL_SERVER_ERROR` for generic errors?

We can return a `NOT_FOUND` response outside of the cluster executor call if the specified `nodeName` is not part of the members list.

Reply:
AFAICT we don't have the name of the members locally, only the addresses :( Previously I created [ISPN-14076](https://issues.redhat.com/browse/ISPN-14076) for that.
I was assuming that if the result is null after the requests then no node has the target name, and left the `INTERNAL_SERVER_ERROR` to the underlying error handler. Do you think that's ok?

Reply:
> AFAICT we don't have the name of the members locally, only the addresses :( Previously I created [ISPN-14076](https://issues.redhat.com/browse/ISPN-14076) for that.

Argh Ok, I didn't realise the discrepancy between the two.



> I was assuming that if the result is null after the requests then no node has the target name, and left the `INTERNAL_SERVER_ERROR` to the underlying error handler. Do you think that's ok?

:+1: